wip-mariocao-blockweight.md

```diff
@@ -0,0 +1,282 @@
+ - **beta** is the return type factor
+
+
+#### Value transfers
+
+The main goal of value transfers is to keep in line the storage of nodes in terms of UXTOs. Therefore, value transfers that join several UTXOs (decreases storage footprint) are favored in detriment of value transfer that split to many UTXOs (increases storage footprint).
+
+The proposed approach is weight significantly more the transaction outputs than the consumed inputs:
+
+`VT_weight = N*INPUT_SIZE + M*OUTPUT_SIZE*gamma`
+
+where:
+ - **VT_weight** is the weight applied to a value transfer transaction
+ - **gamma** is the output multiplicative factor
+ - **INPUT_SIZE** is the size of each input, which is fixed in size
+ - **OUTPUT_SIZE** is the size of each output, fixed in size
+ - **N** is the number of inputs
+ - **M** is the number of outputs
+
+
+#### Rest of transactions
+
+As the rest of transaction will be allocated in the unbounded bucket as there is no need to weight them.
+
+Commit, reveal and tally transactions will be indirectly limited by previously accepted data requests.
+
+Mint transactions are included by the miners are they are limited to 1 per block.
+
+Additionally, by protocol design, commit and reveal transactions cannot be used as mechanism for transferring value as their output addresses must be equal to the node's identity.
+
+
+#### Estimation of transaction lengths
+
+In order to better understand the impact of assigning different weights to different transaction types it is important to analyze the approximate length of each kind of transaction. While some of them have fixed sizes (e.g., Mint transactions) others may depend on e.g., the number inputs/outputs. The table summarizes both the minimum size of each of these transactions and the average size under common number of inputs/outputs.
+
+- estimation of field lengths
+- do not take into consideration serialization overhead
+
+| Transaction Type |  Minimum size  |       Total size  |  Measurements  |
+| ---------------- | -------------- | ----------------- | -------------- |
+| Mint             |       40 bytes |          76 bytes |    ~  40 bytes |
+| Value Transfer   |      133 bytes |  `N*36+M*36+N*97` |    ~ 205 bytes |
+| Data Request     | 133+data bytes |   `36+N*133+data` |    ~ 400 bytes |
+| Commit           |      376 bytes |       `279+N*133` | ~ W\*412 bytes |
+| Reveal           |      149 bytes |      `149+result` | ~ W\*202 bytes |
+| Tally            | 124+data bytes |  `52+W*36+result` |    ~ 105+W\*36 |
+
+where:
+ - *N* refers to the number of inputs,
+ - *M* to the number of outputs, 
+ - *W* to the number of witnesses specified by the data request,
+ - *data* refers to length of the RADON script to be executed by witnesses,
+ - *result* refers to the length of the bytes retrieved by witnesses.
+
+
+### Specification
+
+Based on the proposal, the following parameter definitions are suggested:
+
+- A maximum block weight of 100K
```

Comment:
The implementation ignores this point, is it just a suggestion?

Reply:
Why so? 100K is a weight not a size, thus block size might exceed in real size (for example when including commits and reveals). The weight is distributed among the VTT bucket (20K) and the DR bucket (80K).